### PR TITLE
Improve daterange selection for changes

### PR DESCRIPTION
**Pull request by harriebird**
## Proposed changes

This improves the selection of date ranges for the generation of changes using the `DateRangeField`.
This solves issue #11846.

Example image:
Screenshot 2024-06-19 at 15-40-35 Changes @ Devel Weblate

## Checklist

- [ ] Lint and unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added documentation to describe my feature.
- [ ] I have squashed my commits into logic units.
- [ ] I have described the changes in the commit messages.

### CODE REVIEW

File: weblate/trans/tests/test_reports.py [hunk 1]
```diff
@@ -167,7 +167,11 @@ def test_credits_view_html(self) -> None:
     def test_credits_blank_period(self):
         period = ""
         response = self.get_credits("json", period=period, follow=True)
-        self.assertContains(response, "Error in parameter period: Invalid date!")
+        self.assertEqual(response.status_code, 200)
+        self.assertJSONEqual(
+            response.content.decode(),
+            [{"Czech": [["weblate@example.org", "Weblate <b>Test</b>", 1]]}],
+        )
```
Comment (nijel) [inline weblate/trans/tests/test_reports.py:174]: Reports should not be allowed without a period for performance reasons.

Comment (harriebird) [reply weblate/trans/tests/test_reports.py:172]: Made the `DateRangeField` in the `ReportsForm` required. I also adjusted the `DateRangeField` to disallow empty values when `required` field is set to True.

File: weblate/trans/views/reports.py [hunk 2]
```diff
@@ -83,10 +83,15 @@ def get_credits(request, path=None):
         show_form_errors(request, form)
         return redirect_param(obj or "home", "#reports")
 
+    start_date = end_date = None
+    if form.cleaned_data["period"]:
+        start_date = form.cleaned_data["period"]["start_date"]
+        end_date = form.cleaned_data["period"]["end_date"]
+
```
Comment (nijel) [inline weblate/trans/views/reports.py:90]: Please revert the changes here as well, reports need period to be provided.

Comment (harriebird) [reply weblate/trans/views/reports.py]: Reverted the changes for checking if the `form.cleaned_data["period"]` is None.

**Comment by nijel** [comment]
This introduced regression: https://github.com/WeblateOrg/weblate/issues/11954
